### PR TITLE
New package: oracle-jdk-11-11.0.2

### PR DESCRIPTION
**Pull request by SevereOverfl0w**
I think this is mostly complete.  How do I handle the icons and the .desktop files? They will conflict with the oracle-jdk package.  One option is to make that part of the alternatives, another is to create an oracle-jdk-common package for both to depend on.

### CODE REVIEW

**Comment by Hoshpak** [comment]
```
Further, You may not:
-    make the Programs available in any manner to any third party (other than Contractors acting on Your behalf as set forth in this Agreement);
```
https://www.oracle.com/technetwork/java/javase/terms/license/javase-license.html
Seems non-distributable to me.

**Comment by SevereOverfl0w** [comment]
This is no different to the existing oracle-jdk package, so we should remove that also? Also, this isn't a redistribution (the distfile isn't contained at all), it's downloaded at runtime by the user.

**Comment by Hoshpak** [comment]
Alright then, I just saw that the package downloaded the JDK and assumed it would be packaged.

**Comment by SevereOverfl0w** [comment]
I've been using this as my regular driver, and it works very well. The big remaining step is the xbps-alternatives setting for the auxiliary files like the profile.sh and the icons.

Switching between 8 & 11 is fairly seamless, but the JAVA_HOME setting sometimes needs updating, but that can be fixed as part of the same changes for set up.

**Comment by Francesco149** [comment]
would love for this to be merged, I'm thinking of packaging ghidra which requires jdk 11

File: srcpkgs/oracle-jdk-11/template [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+homepage="http://www.oracle.com/technetwork/java/javase/downloads/index.html"
+license="oracle"
+lib32disabled=yes
+repository="nonfree"
+only_for_archs="x86_64"
+depends="hicolor-icon-theme desktop-file-utils xdg-utils shared-mime-info wget"
+hostmakedepends="wget"
+provides="java-environment-${version}_1"
+_filename="jdk-11_linux-x64_bin.tar.gz"
+distfiles="http://download.oracle.com/otn-pub/java/jdk/${version}/${_tag}/jdk-${version}_linux-x64_bin.tar.gz"
```
Comment (Oreo639) [inline srcpkgs/oracle-jdk-11/template]: I believe correct url is `http://download.oracle.com/otn-pub/java/jdk/${_longVersion}/${_tag}/jdk-${version}_linux-x64_bin.tar.gz`

File: srcpkgs/oracle-jdk-11/template [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+checksum="7b4fd8ffcf53e9ff699d964a80e4abf9706b5bdb5644a765c2b96f99e3a2cdc8"
+wrksrc="jdk-$_shortVersion"
```
Comment (Oreo639) [inline srcpkgs/oracle-jdk-11/template]: I don't see a `_shortVersion` variable. I think you mean `$version`

File: srcpkgs/oracle-jdk-11/template [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+		$distfiles
+	echo "$checksum  $_filename" | sha256sum -c || { sha256sum "$_filename"; return 1; }
+}
+
+do_extract() {
+	tar xf $_filename -C ..
+}
+
+do_build() {
+	sed \
```
Comment (Piraty) [inline srcpkgs/oracle-jdk-11/template]: please `vsed` to detect unchanged files

File: srcpkgs/oracle-jdk-11/template [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+		-e "s|%LONG_VERSION%|${longVersion}|" \
```
Comment (Piraty) [inline srcpkgs/oracle-jdk-11/template]: typo: `_longVersion`

File: srcpkgs/oracle-jdk-11/template [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+	vinstall oracle-jdk-11-vars.sh 644 "usr/lib/jvm/"
+
+	vlicense legal/java.base/LICENSE
+
+	# vinstall $FILESDIR/java-policy-settings.desktop 644 usr/share/applications
+	ln -s jdk-${version} ${DESTDIR}/usr/lib/jvm/oracle-jdk
+
+	# for i in 16 20 22 24 32 40 48 64 72 96 128 256 512; do
+	# 	vinstall $FILESDIR/oracle-jdk-${i}x${i}.png 644 usr/share/icons/hicolor/${i}x${i}/apps oracle-java.png
+	# done
```
Comment (Piraty) [inline srcpkgs/oracle-jdk-11/template]: Is it really necessary to provide the icons in the tree?
This is commented...?

**Comment by Anachron** [comment]
Is this still being updated?

**Comment by SevereOverfl0w** [comment]
I think someone has now released 11? This can probably be closed.

**Comment by Anachron** [comment]
No, where?

**Comment by Francesco149** [comment]
pretty sure openjdk11 is already merged into void. we even have ghidra now

**Comment by Anachron** [comment]
Oh its called `openjdk11` and not `open-jdk-11`. Github search doesn't find that. This can be closed.

**Pull request closed without merging**